### PR TITLE
WIP: Duration picker

### PR DESCRIPTION
**Pull request by Gustl22**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

#144536

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: dev/integration_tests/flutter_gallery/lib/duration_main.dart [hunk 1]
```diff
@@ -0,0 +1,127 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
```
Comment (Piinks) [inline dev/integration_tests/flutter_gallery/lib/duration_main.dart:1]: This is in the integration test directory, is there a test that makes use of this? Maybe this should be in the examples directory instead?

**Review by Piinks — COMMENTED** [review]
Hi @Gustl22, thanks for contributing! It looks like this is currently failing some tests, missing some tests, but also marked as a draft, would you like to continue working on this change?

If so, is this widget part of the material design specification? We typically use the tokens provided by Material Design to configure widgets in the library.

File: dev/integration_tests/flutter_gallery/lib/duration_main.dart (hunk 1 above)
Comment (Gustl22) [reply dev/integration_tests/flutter_gallery/lib/duration_main.dart:1]: This is only for a evaluating. Not sth which lands in production :)

**Comment by Gustl22** [comment]
Hi @Piinks,
thanks for jiming in! I'd love to continue working on this. But before continuing this PR (and spent hours/days on tests etc.) I'd like to ensure, that this feature actually meets the design requirements. I mainly asked these questions already here:
https://github.com/flutter/flutter/issues/144536#issuecomment-2039137635
and on [Discord](https://discord.com/channels/608014603317936148/608018585025118217/1235253118468358154)
- Does this feature / design need to be validated first

To answer your question:
- This Widget is NOT specified by the material design specification, but it can be implemented very similar to the [time picker](https://m3.material.io/components/time-pickers/overview)

We can also close this until the flutter / material team decided, if this feature request is valid. I don't know the procedure for this actually.
Thx :)

**Comment by Gustl22** [comment]
Closing, as it's not in  the official specs of Material. May will publish this as a dedicated package. Although it would be good to resolve #131247 first.

**Comment by Gustl22** [comment]
FYI: I published the package here: https://pub.dev/packages/material_duration_picker
GH: https://github.com/Oberhauser-Dev/dart_packages/tree/main/packages/material_duration_picker
It's still kept in sync with the [feature branch](https://github.com/Gustl22/flutter/tree/144536-duration-picker) I created for the PR, so merging it into Flutter is still an option.

**Pull request closed without merging**